plugins/cassandra/cassandrawu.cpp

```diff
@@ -3554,14 +3554,7 @@ class CCasssandraWorkUnitFactory : public CWorkUnitFactory, implements ICassandr
                     merger->addResult(*new CassandraResult(fetchDataForWuid(workunitInfoMappings, value, true)));
                 }
             }
-            else if (!thorTimeThreshold.isEmpty())
-            {
-                merger->addPostFilters(goodFilters, 0);
-                merger->addPostFilters(poorFilters, 0);
-                merger->addPostFilters(remoteWildFilters, 0);
-                merger->addResult(*new CassandraResult(fetchDataByThorTime(thorTimeThreshold, sortDescending, merger->hasPostFilters() ? 0 : pageSize+startOffset)));
-            }
-            else if (sortByThorTime)
+            else if (sortByThorTime || !thorTimeThreshold.isEmpty())
```

Comment:
I notice (on line 3435) that thorTimeThreshold is cleared if !sortByThorTime ..

So still not sure about the logic here.
Has it been tested?

@wangkx

Reply:
The code is used for the pages after the first page (line 3423) and there is WU inside the cache (line 3427). Looks like that the code is to clear old thorTimeThreshold. After the code, the thorTimeThreshold is set as needed at line 3507/3508.  I am going to move line 3435 after line 3436.